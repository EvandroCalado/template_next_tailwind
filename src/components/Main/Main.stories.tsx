import { Meta, StoryFn } from '@storybook/react';
import { Main, MainProps } from './Main';

export default {
  title: 'Components/Main',
  component: Main,
  args: {
    title: 'template next',
  },
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const Default: StoryFn<MainProps> = (args) => <Main {...args} />;
