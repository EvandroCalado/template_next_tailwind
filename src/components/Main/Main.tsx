import Image from 'next/image';
import { FC } from 'react';

export interface MainProps {
  title?: string;
}

export const Main: FC<MainProps> = ({ title = 'template next' }) => {
  return (
    <main className="flex h-screen w-full flex-col items-center justify-center gap-4 bg-zinc-100 px-4 md:flex-row lg:gap-20 dark:bg-zinc-800">
      <div className="flex flex-col items-center justify-center space-y-4 md:space-y-8">
        <h1 className="text-xl font-bold capitalize text-zinc-800 md:text-4xl lg:text-6xl dark:text-zinc-100">
          {title}
        </h1>

        <Image
          src="./logo.svg"
          alt="desenvolvedor em frente a um computador"
          width={400}
          height={400}
          priority
        />
      </div>

      <div className="flex flex-col items-center justify-center space-y-4 md:space-y-8">
        <h4 className="text-xl font-bold capitalize text-zinc-800 md:text-4xl lg:text-6xl dark:text-zinc-100">
          tecnologias
        </h4>

        <ul className="text-sm text-zinc-800 md:space-y-2 md:text-base dark:text-zinc-100">
          <li>
            <span className="font-bold text-emerald-300">🐳 Docker: </span>{' '}
            Plataforma para desenvolvimento, envio e execução de aplicações em
            contêineres.
          </li>
          <li>
            <span className="font-bold text-emerald-300">🚀 Next.js:</span>{' '}
            Framework React para desenvolvimento de aplicações web.
          </li>
          <li>
            <span className="font-bold text-emerald-300">⚙️ TypeScript:</span>{' '}
            Superset do JavaScript que adiciona tipagem estática ao código.
          </li>
          <li>
            <span className="font-bold text-emerald-300">💄 Tailwind CSS:</span>{' '}
            Framework de CSS utilitário para estilização rápida e consistente.
          </li>
          <li>
            <span className="font-bold text-emerald-300">📖 Storybook:</span>{' '}
            Ferramenta para desenvolvimento e documentação de componentes de
            interface.
          </li>
          <li>
            <span className="font-bold text-emerald-300">🧪 Vitest:</span>{' '}
            Framework de testes rápidos para projetos modernos de frontend.
          </li>
          <li>
            <span className="font-bold text-emerald-300">🔧 Husky:</span>{' '}
            Ferramenta para adicionar ganchos de Git facilmente.
          </li>
          <li>
            <span className="font-bold text-emerald-300">🛠️ Lint-Staged:</span>{' '}
            Executa linters em arquivos modificados no repositório.
          </li>
          <li>
            <span className="font-bold text-emerald-300">📝 ESLint:</span>{' '}
            Ferramenta de linting para identificar e corrigir problemas no
            código JavaScript.
          </li>
          <li>
            <span className="font-bold text-emerald-300">📑 Prettier:</span>{' '}
            Ferramenta de formatação de código.
          </li>
        </ul>
      </div>
    </main>
  );
};
