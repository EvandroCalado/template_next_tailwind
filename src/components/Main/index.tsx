import Image from 'next/image';
import { FC } from 'react';

export interface MainProps {
  title?: string;
}

export const Main: FC<MainProps> = ({ title = 'template next' }) => {
  return (
    <main className="flex h-screen w-full flex-col items-center justify-center space-y-8 bg-zinc-800 px-4">
      <h1 className="text-4xl font-bold capitalize text-zinc-100 sm:text-6xl">
        {title}
      </h1>

      <h4 className="text-sm capitalize text-emerald-300 sm:text-xl">
        com typescript, tailwind, storybook e vitest
      </h4>

      <Image
        src="./logo.svg"
        alt="desenvolvedor em frente a um computador"
        width={400}
        height={400}
        priority
      />
    </main>
  );
};
