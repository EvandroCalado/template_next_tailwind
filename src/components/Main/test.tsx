import { render, screen } from '@testing-library/react';
import { Main } from '.';

describe('Main', () => {
  it('should render correctly', () => {
    render(<Main />);

    const main = screen.getByRole('heading', { name: /template next/i });
    const img = screen.getByRole('img', { name: /desenvolvedor/i });

    expect(main).toBeInTheDocument();
    expect(img).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
