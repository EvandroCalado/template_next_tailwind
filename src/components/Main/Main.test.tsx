import { render, screen } from '@testing-library/react';
import { Main } from './Main';

describe('Main', () => {
  it('should render correctly', () => {
    render(<Main />);

    const main = screen.getByRole('heading', { name: /template next/i });
    const img = screen.getByRole('img', { name: /desenvolvedor/i });

    expect(main).toBeInTheDocument();
    expect(img).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
