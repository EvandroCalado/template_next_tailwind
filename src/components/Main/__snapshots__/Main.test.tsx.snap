// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main > should render a snapshot 1`] = `
<main
  class="flex h-screen w-full flex-col items-center justify-center gap-4 bg-zinc-100 px-4 md:flex-row lg:gap-20 dark:bg-zinc-800"
>
  <div
    class="flex flex-col items-center justify-center space-y-4 md:space-y-8"
  >
    <h1
      class="text-xl font-bold capitalize text-zinc-800 md:text-4xl lg:text-6xl dark:text-zinc-100"
    >
      template next
    </h1>
    <img
      alt="desenvolvedor em frente a um computador"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="400"
      src="./logo.svg"
      style="color: transparent;"
      width="400"
    />
  </div>
  <div
    class="flex flex-col items-center justify-center space-y-4 md:space-y-8"
  >
    <h4
      class="text-xl font-bold capitalize text-zinc-800 md:text-4xl lg:text-6xl dark:text-zinc-100"
    >
      tecnologias
    </h4>
    <ul
      class="text-sm text-zinc-800 md:space-y-2 md:text-base dark:text-zinc-100"
    >
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          🐳 Docker: 
        </span>
         
        Plataforma para desenvolvimento, envio e execução de aplicações em contêineres.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          🚀 Next.js:
        </span>
         
        Framework React para desenvolvimento de aplicações web.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          ⚙️ TypeScript:
        </span>
         
        Superset do JavaScript que adiciona tipagem estática ao código.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          💄 Tailwind CSS:
        </span>
         
        Framework de CSS utilitário para estilização rápida e consistente.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          📖 Storybook:
        </span>
         
        Ferramenta para desenvolvimento e documentação de componentes de interface.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          🧪 Vitest:
        </span>
         
        Framework de testes rápidos para projetos modernos de frontend.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          🔧 Husky:
        </span>
         
        Ferramenta para adicionar ganchos de Git facilmente.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          🛠️ Lint-Staged:
        </span>
         
        Executa linters em arquivos modificados no repositório.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          📝 ESLint:
        </span>
         
        Ferramenta de linting para identificar e corrigir problemas no código JavaScript.
      </li>
      <li>
        <span
          class="font-bold text-emerald-300"
        >
          📑 Prettier:
        </span>
         
        Ferramenta de formatação de código.
      </li>
    </ul>
  </div>
</main>
`;
